### PR TITLE
chore: use renovate to bump flagd version

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR

- removes script and pipelines checking.bumping flagd version
- introduces renovate regex to automatically bump flagd version

### Related Issues

Part of #370

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
   },
   "regexManagers": [
     {
-      "fileMatch": ["^Makefile$"],
-      "matchStrings": ["FLAGD_VERSION=(?<currentValue>.*?)\\n"],
+      "fileMatch": ["^apis/core/v1alpha1/flagsourceconfiguration_types.go$"],
```
[the remark this answers is not in the document]
Comment (odubajDT) [reply renovate.json:18]: sure

[the remark this answers is not in the document]
Comment (odubajDT) [reply renovate.json]: adapted

File: .github/workflows/release-please.yml
Comment (beeme1mr) [inline .github/workflows/release-please.yml:1]: Please also remove the `update-flagd` command on lines 71-73.

Comment (odubajDT) [reply .github/workflows/release-please.yml:1]: good catch, fixed

**Review by beeme1mr — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**